app/assets/javascripts/discourse/app/services/presence.js

```diff
@@ -606,7 +606,7 @@ export default class PresenceService extends Service {
       );
     } else if (
       !this._nextUpdateTimer &&
-      this._presentChannels.length > 0 &&
+      this._presentChannels.size > 0 &&
```

Comment:
`this._presentChannels` is a JS `Set`, so [`size`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/size) should always return a value. What makes you say it's 'always undefined'? 🤔

Reply:
Yes that's what I said. `.length` is always undefined `.size` works. ( right now in the code it's `.length` ) I was trying to implement same mechanism and I am using typescript so it complained and I tried it in the console:


<img width="370" alt="image" src="https://github.com/discourse/discourse/assets/20217427/908f28ab-efed-4c54-9465-5302b1e3dc50">

Reply:
My pleasure!